clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1005,10 +1025,31 @@ public SelectorMetrics(Metrics metrics, String metricGrpPrefix, Map<String, Stri
             this.successfulAuthentication.add(createMeter(metrics, metricGrpName, metricTags,
                     "successful-authentication", "connections with successful authentication"));
 
+            this.successfulReauthentication = sensor("successful-reauthentication:" + tagsSuffix);
+            this.successfulReauthentication.add(createMeter(metrics, metricGrpName, metricTags,
+                    "successful-reauthentication", "successful re-authentication of connections"));
+
+            String successfulV0AuthenticationMetricTagKey = "auth";
+            String successfulV0AuthenticationMetricTagValue = "v0";
+            this.successfulV0Authentication = sensor("successful-v0-authentication:" + tagsSuffix
```

Comment:
Seeing it here, the metric name doesn't look right. `v0` refers to SaslAuthenticateRequest version. Perhaps `sasl-authentication` with tag `version=0` or something similar to our request metrics which has apikey and version would be better?

Reply:
@rajinisivaram I realized today that we also have to count the connections from older clients that do not send SASL_AUTHENTICATE.  So it might be good to not refer to "version" at all.  Maybe it is best to not refer to "sasl" either?  How about `successful-authentication-no-reauth` as the metric name?  Then maybe we don't need a special tag?  If we still do need a tag then maybe `reauthSupport=false` to avoid referring to "version"?

Reply:
Changing to `successful-authentication-no-reauth-total` without extra tags.